resources/metric.mjs

```diff
@@ -0,0 +1,90 @@
+import * as Statistics from "./statistics.mjs";
+
+/** Number of milliseconds in a single minute. */
+export const MILLIS_PER_MIN = 60 * 1000;
+
+export class Metric {
+    /** @param {string} name */
+    constructor(name, unit = "ms") {
+        if (typeof name !== "string")
+            throw new Error(`Invalid metric.name=${name}, expected string.`);
+        this.name = name;
+        this.unit = unit;
+
+        this.mean = 0.0;
+        this.geomean = 0.0;
+        this.delta = 0.0;
+        this.percentDelta = 0.0;
+
+        this.sum = 0.0;
+        this.min = 0.0;
+        this.max = 0.0;
+
+        /** @type {number[]} */
```

Comment:
I don't think we should have these comments.

Reply:
- I will remove them (that's literally just copy-pasted from the grandprix repo)
- Might be something for the future though, the type annotations are quite nice when working with vscode (and we could have typescript verification)